app2.py

```diff
@@ -0,0 +1,264 @@
+        self.server = SERVER
+        self.group = 'administrators'
+        self.url = f'https://{self.server}.jfrog.io/artifactory/api/'
+        self._user_credentials() # TODO: activate this method after testing
+        self.token = self._get_token_for_group(self.group)
+        self.session = requests.Session()
+        self._set_session(self.token)
+        # self.menu()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
+
+    def _user_credentials(self) -> None:
+        # Get user credentials from the command line on script run
+        parser = argparse.ArgumentParser(description='CLI for the JFrog REST API')
+        parser.add_argument('-u', '--username', help="personal username to login to JFrog's REST API", required=True)
+        parser.add_argument('-p', '--password', help="personal password to login to JFrog's REST API", required=True)
+        args = parser.parse_args()
+        # args, unknown = parser.parse_known_args()
+        self.username = args.username
+        self.password = args.password
+
+    #----- Utility methods -----
+    def _set_session(self, token: str) -> None:
+        self.headers = {'Authorization': f'Bearer {token}'}
+        self.session.headers.update(self.headers)
+
+    def _set_url(self, endpoint: str) -> str:
+        url = ''.join([self.url, endpoint])
+        return url
+
+    def _get_token_for_group(self, group: str) -> str:
+        # curl -u $user:$password -XPOST "https://artitestserver.jfrog.io/artifactory/api/security/token" -d "username=$user" -d "scope=member-of-groups:readers" > .tok3
+        
+        endpoint = 'security/token'
+        # url = ''.join([self.url, endpoint])
+        url = self._set_url(endpoint)
+        data = {'username':self.username, 'password':self.password, 'scope=member-of-groups':group}
+        r = requests.post(url, auth=(self.username, self.password), data=data)
+        # r = self.session.post(url, auth=(self.username, self.password), data=self.data)
+        return r.json()['access_token']
+
+    #----- Menu options implementation -----
+    # 1
+    def _system_ping(self) -> requests.Response:
+        # curl -H "Authorization: Bearer $Access_Token" https://artitestserver.jfrog.io/artifactory/api/system/ping
+        endpoint = 'system/ping'
+        url = self._set_url(endpoint)        
+        r = self.session.get(url, headers=self.session.headers)
+        return r
+
+    # 2
+    def _system_version(self) -> str:
+        # curl -H "Authorization: Bearer $Access_Token" https://artitestserver.jfrog.io/artifactory/api/system/ping
+        endpoint = 'system/version'
+        url = self._set_url(endpoint)    
+        r = self.session.get(url, headers=self.session.headers)
+        return r.json()['version']
+
+    # 3
+    def _create_user(self) -> requests.status_codes:
+        # Notes: Requires Artifactory Pro
+        # Usage: PUT /api/security/users/{userName}
+        username = input("Enter username: ")
+        email = input("Enter email: ")
```

Comment:
I would implement a `_validate_email()` function to verify the email is in the right format.
For example: `XXX@YYY.com` is good but `XXX@YYY` is not